tsdb/head.go

```diff
@@ -1021,6 +1021,15 @@ func (a *initAppender) AddFast(ref uint64, t int64, v float64) error {
 	return a.app.AddFast(ref, t, v)
 }
 
+func (a *initAppender) GetRef(lset labels.Labels) (uint64, bool) {
+	if g, ok := a.app.(interface {
+		GetRef(lset labels.Labels) (uint64, bool)
```

Comment:
I would suggest being a bit more explicit and put this single method `GetRef`  interface next to `Appender` in `storage/interface.go` with comment that it is an interface optionally supported by Appender. 🤗

Reply:
Maybe also add `var _ optionalGetRefInterface = initAppender{}` so you get notified if you break it.

Reply:
Done.

Reply:
Let's change this code then